src/main/java/com/coherentsolutions/pot/insuranceservice/service/InsurancePackageManagementService.java

```diff
@@ -44,6 +48,16 @@ private void validateOnUpdate(InsurancePackage insurancePackage,
     }
   }
 
+  private void validateAndSetPlans(InsurancePackageDto insurancePackageDto, InsurancePackage insurancePackage) {
+    if (insurancePackageDto.getPlanIds() != null && !insurancePackageDto.getPlanIds().isEmpty()) {
```

Comment:
NIT: You can use Spring's `CollectionUtils` to simplify the code a little:
```suggestion
    if (!CollectionUtils.isEmpty(insurancePackageDto.getPlanIds())) {
```